pkg/handler/aws/uploadProfilePicture/main.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/CollActionteam/collaction_backend/auth"
+	"github.com/CollActionteam/collaction_backend/internal/uploads"
+	awsRepository "github.com/CollActionteam/collaction_backend/pkg/repository/aws"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+func handler(ctx context.Context, req events.APIGatewayV2HTTPRequest) (res events.APIGatewayV2HTTPResponse, err error) {
+	usrInf, err := auth.ExtractUserInfo(req)
+	if err != nil {
+		res = events.APIGatewayV2HTTPResponse{StatusCode: http.StatusForbidden, Body: "user not authorized"}
+		return res, err
+	}
+
+	userID := usrInf.UserID()
+	sess := session.Must(session.NewSession())
+	profileImageUploadRepo := awsRepository.NewProfilePicture(sess)
+
+	strUrl, err := uploads.NewProfileImageUploadService(profileImageUploadRepo).GetUploadUrl(ctx, "png", userID)
+	if err != nil {
+		res = events.APIGatewayV2HTTPResponse{StatusCode: http.StatusInternalServerError, Body: "error generating link"}
+		return res, err
+	}
+
+	response, _ := json.Marshal(map[string]interface{}{"upload_url": strUrl})
+
+	return events.APIGatewayV2HTTPResponse{
+		StatusCode: http.StatusOK,
+		Body:       string(response),
+	}, nil
```

Comment:
please use the utils functions here for the response. Check utils/api.go

Reply:
I followed the newer convention of the refactoring process so I am not sure if we would still need that because the previously merged PR @mrsoftware  serving as template had this format. So if we can discuss as a team we can have more clarity on these. @rubenhorn

Reply:
I think we have not explicitly made a decision regarding this.  
One one hand, using utility functions provides consistency and reduces the risk of coding errors, on the other hand however, it also introduces more dependencies between packages.  
I also think we should discuss this (perhaps in one of the weekly backend meetings 😉), but it should not matter for this PR and we should instead make a separate refactoring task to go through the code and implement our decision.

Reply:
Alright 👍